src/Shared/Microsoft.DotNet.Kusto/KustoServiceCollectionExtensions.cs

```diff
@@ -9,6 +9,10 @@ namespace Microsoft.DotNet.Kusto
 {
     public static class KustoServiceCollectionExtensions
     {
+        public static IServiceCollection AddKustoClientProvider(this IServiceCollection services)
+        {
+            return services.AddSingleton<IKustoClientProvider, KustoClientProvider>();
+        }
```

Comment:
Add a blank line.

Who's calling this without configuring the options? That seems... bad.  We only have the existing overload so that you don't forget to configure it... providing the optionless overload removes that safety.

I'd much rather only have the 1 overload.

Reply:
Is the problem we don't have a method that allows us fetch the Configuration?  We could probably add that overload, one that takes Action<IConfigurationRoot, KustoClientProviderOptions> (it's just a call to services.GetService<IConfigurationRoot>(), I think)

---
In reply to: [384084555](https://github.com/dotnet/arcade-services/pull/990#discussion_r384084555) [](ancestors = 384084555)

Reply:
Yeah, it was that we don't have the method to fetch Configuration. I wasn't sure if it was better to go that way or this way.

Reply:
Ok, I remember why I did it this way: I couldn't figure out how to get the options out of the Action, and it was just easier to follow a model I already had (we do this for GitHubOptions and AzureDevOpsTokenProvider. I will try to go find docs/an example in our code where we pass both IConfigurationRoot and Options.